include/exadg/incompressible_navier_stokes/spatial_discretization/spatial_operator_base.cpp

```diff
@@ -147,10 +145,14 @@ SpatialOperatorBase<dim, Number>::fill_matrix_free_data(
   matrix_free_data.insert_constraint(&constraint_u_scalar, field + dof_index_u_scalar);
 
   // quadrature
-  matrix_free_data.insert_quadrature(dealii::QGauss<1>(param.degree_u + 1), field + quad_index_u);
-  matrix_free_data.insert_quadrature(dealii::QGauss<1>(param.get_degree_p(param.degree_u) + 1),
+  matrix_free_data.insert_quadrature(fe_u->reference_cell().template get_gauss_type_quadrature<dim>(
+                                       param.degree_u + 1),
+                                     field + quad_index_u);
+  matrix_free_data.insert_quadrature(fe_p->reference_cell().template get_gauss_type_quadrature<dim>(
+                                       param.get_degree_p(param.degree_u) + 1),
                                      field + quad_index_p);
-  matrix_free_data.insert_quadrature(dealii::QGauss<1>(param.degree_u + (param.degree_u + 2) / 2),
+  matrix_free_data.insert_quadrature(fe_u->reference_cell().template get_gauss_type_quadrature<dim>(
+                                       param.degree_u + (param.degree_u + 2) / 2),
```

Comment:
Increased the `dim` of the quadrature from 1 to dim, keeping this in mind: https://github.com/exadg/exadg/pull/305#discussion_r1082353641